SlowCheetah.VisualStudio/SlowCheetahPackage.cs

```diff
@@ -671,144 +580,35 @@ private bool ProjectSupportsTransforms(IVsProject project)
         /// <returns>True if the item supports transforms</returns>
         private bool ItemSupportsTransforms(IVsProject project, uint itemid)
         {
-            string itemFullPath = null;
-
+            string itemFullPath;
             if (ErrorHandler.Failed(project.GetMkDocument(itemid, out itemFullPath)))
             {
                 return false;
             }
 
             bool itemSupportsTransforms = false;
-            // make sure its not a transform file itself
-            bool isTransformFile = IsItemTransformItem(project, itemid);
-
-
             FileInfo transformFileInfo = new FileInfo(itemFullPath);
+            // make sure its not a transform file itself
             bool isWebConfig = string.Compare("web.config", transformFileInfo.Name, StringComparison.OrdinalIgnoreCase) == 0;
+            bool isTransformFile = IsItemTransformItem(project, itemid);
+            bool isExtensionSupportedForFile = PackageUtilities.IsExtensionSupportedForFile(itemFullPath);
+            bool isXmlFile = PackageUtilities.IsXmlFile(itemFullPath);
```

Comment:
Inlining all of them into the if would be hard to read and inlining some but not all seems messy